modules/tinymce/src/core/main/ts/content/NbspTrim.ts

```diff
@@ -5,62 +5,31 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
-import { Unicode } from '@ephox/katamari';
-import * as NodeType from '../dom/NodeType';
-
-const isAfterNbsp = (container: Node, offset: number) => NodeType.isText(container) && container.nodeValue[offset - 1] === Unicode.nbsp;
-
-const trimOrPadLeftRight = (rng: Range, html: string): string => {
-  const container = rng.startContainer;
-  const offset = rng.startOffset;
-
-  const hasSiblingText = function (siblingName) {
-    return container[siblingName] && container[siblingName].nodeType === 3;
-  };
-
-  if (NodeType.isText(container)) {
-    if (offset > 0) {
-      html = html.replace(/^&nbsp;/, ' ');
-    } else if (!hasSiblingText('previousSibling')) {
-      html = html.replace(/^ /, '&nbsp;');
-    }
-
-    if (offset < container.length) {
-      html = html.replace(/&nbsp;(<br>|)$/, ' ');
-    } else if (!hasSiblingText('nextSibling')) {
-      html = html.replace(/(&nbsp;| )(<br>|)$/, '&nbsp;');
-    }
+import { SugarElement } from '@ephox/sugar';
+import DOMUtils from '../api/dom/DOMUtils';
+import CaretPosition from '../caret/CaretPosition';
+import { needsToBeNbspLeft, needsToBeNbspRight } from '../keyboard/Nbsps';
+
+const trimOrPadLeftRight = (dom: DOMUtils, rng: Range, html: string): string => {
+  const root = SugarElement.fromDom(dom.getRoot());
+
+  // Adjust the start if it needs to be an nbsp
+  if (needsToBeNbspLeft(root, CaretPosition.fromRangeStart(rng))) {
+    html = html.replace(/^ /, '&nbsp;');
+  } else {
+    html = html.replace(/^&nbsp;/, ' ');
```

Comment:
It seems suspicious that you can just write `&nbsp;` inside a regex and have it work, but I guess if it works it works. Is there seriously no escaping needed on the `&`s?

Reply:
Ampersand isn't a special character in a regex, so that's fine. Also remember that it's matching an actual string here, not 1 single character.